telegram/_message.py

```diff
@@ -1546,6 +1588,7 @@ async def reply_text(
         message_thread_id: Optional[int] = None,
         link_preview_options: ODVInput["LinkPreviewOptions"] = DEFAULT_NONE,
         reply_parameters: Optional["ReplyParameters"] = None,
+        business_connection_id: Optional[str] = None,
```

Comment:
> since we just had the topic with `message_thread_id`: I'm wondering if we should pass `self.business_connection_id` to `send_message` by default. 

I didn't follow that discussion. But we shouldn't pass business_connection_id in every request because the bot may not send every message on behalf of the business account, right? And yes the bot needs permission to send messages.

> I imagine that for the majority of use cases a business will implement a custom bot such that the bot will only ever use one fixed business_connection_id.

One bot may manage the business accounts of multiple people in a company? Anyway it probably still makes sense to add it to Defaults (later) for the other use cases.

> he behavior for reply_text should best be fixed already in the first release for API 7.2.

what is broken?

Reply:
> > since we just had the topic with `message_thread_id`: I'm wondering if we should pass `self.business_connection_id` to `send_message` by default.
> 
> I didn't follow that discussion. But we shouldn't pass business_connection_id in every request because the bot may not send every message on behalf of the business account, right? And yes the bot needs permission to send messages.

See #4139. I agree that bcid shouldn't usually be passed on *every* message. My point here would be that if `Message.bcid` is set, then my guess would be that `Message.reply_text` should pass `send_message(…, bcid=Message.bcid)`.
Regarding write permissions, I now think that this actually should not be a problem: The bot can check the write permission status via `get_business_connection`. If the bot does have write permissions, then it's very benefitial if `repy_text` already passes the bcid. If the bot does *not* have write permissions, I don't think trying to reply makes any sense at all - this would try to send a message to the user from the bots own account, but there is no knowing if the user has contacted the bot yet (without detour over the business account).
Punshline: I think we should pass bcid in reply_text.

> > I imagine that for the majority of use cases a business will implement a custom bot such that the bot will only ever use one fixed business_connection_id.
> 
> One bot may manage the business accounts of multiple people in a company? Anyway it probably still makes sense to add it to Defaults (later) for the other use cases.

The company would still only have a single business account, I imagine …
 
> > he behavior for reply_text should best be fixed already in the first release for API 7.2.
> 
> what is broken?

nothing, I just want to avoid a situation were we want to change behavior in a potentionally breaking way :)

Reply:
Ah I see the point now. The next question would be if  `bcid` is provided as an argument in `reply_text` then?

Reply:
> The next question would be if `bcid` is provided as an argument in `reply_text` then?

Good point, I think we can skip that from the signature of `reply_*`. If users need to customize/override that at some point, we can add it later on and then handle it similar to what we do for reply_parameters or message_thread_id

> And then shouldn't we also do the same for the other message methods which accept bcid?

I would add that for all shortcut methods of the `Message` class where the underlying bot method has such a parameter, yes